docs/source/teams/pluggable_auth.md

```diff
@@ -0,0 +1,683 @@
+system with two modes (`legacy` and `internal`). Legacy mode uses Auth0.
+Internal mode eliminates the Auth0 external dependency and may run in
+environments without egress to the internet. CAS provides a UI, REST API, and
+JavaScript (JS) Hook mechanism to manage FiftyOne Teams user data and
+authentication.
+
+## FiftyOne Authentication Modes
+
+The setting `FIFTYONE_AUTH_MODE` specifies the authentication mode `legacy` or
+`internal`.
+
+(legacy-mode)=
+
+### Legacy Mode
+
+In legacy mode, FiftyOne Teams uses Auth0 for user authentication and
+authorization. This mode requires an external connection to Auth0 endpoints.
+User data is eventually consistent (where changes are reflected across
+FiftyOneTeams eventually). Auth0 contains the configuration for identity
+providers and the persistence of user data. Auth0 supports multiple providers
+(including SAML). For the supported IdPs, see
+[Auth0 Enterprise Identity Providers](https://auth0.com/docs/authenticate/identity-providers/enterprise-identity-providers).
+
+(internal-mode)=
+
+### Internal Mode
+
+In internal mode, FiftyOne Teams the CAS replaces Auth0. FiftyOne Teams will
+not require network egress to external services. User data is immediately
+consistent (where changes are reflected across FiftyOne Teams instantly).
+Directory data is immediately written to MongoDB, and organizations have the
+autonomy to manage their Identity Provider Configuration. Internal mode
+supports
+[OpenID Connect (OIDC)](https://openid.net/developers/discover-openid-and-openid-connect/)
+and [OAuth2](https://oauth.net/2/).
+
+> **NOTE**: SAML is not supported in [internal mode](#internal-mode)
+
+<!-- TODO: Enumerate what connection types are supported. -->
+
+> NOTE: invitations to onboard users are not supported in internal mode.
+
+(super-admin-ui)=
+
+## Super Admin UI
+
+![Super Admin UI](/images/teams/cas/cas_api_docs.png)
+
+The Super Admin UI contains FiftyOne Teams deployment wide configurations. When
+logging into FiftyOne Teams as an admin, you are in the context of an
+organization. Settings are scoped by organization and only apply to that
+organization. The Super Admin UI allows you to administer all organizations and
+global configurations (Identity Providers, Session timeouts, and JS hooks).
+
+> **NOTE**: In v1.6.0, the Super Admin UI is only available in
+> [internal mode](#internal-mode)
+
+![Sign In to CAS](/images/teams/cas/cas_sign_in.png)
+
+To login to this application navigate to
+`https://<YOUR_FIFTYONE_TEAMS_URL>/cas/configurations`. In the top right of the
+screen, and provide the `FIFTYONE_AUTH_SECRET` to login.
+
+> **NOTE**: The value of `FIFTYONE_AUTH_SECRET` should be set prior to
+> installation or upgrade.
+
+(identity-providers)=
+
+## Identity Providers (IdP)
+
+In [internal mode](#internal-mode), use the CAS REST API or Super Admin UI to
+configure FiftyOne teams to authenticate users via OIDC or OAuth2 compatible
```

Comment:
I am under the impression that OAuth2 is a protocol for authorization, not authentication.  Should we update this to be 

```suggestion
configure FiftyOne teams to authenticate users via OIDC compatible
```

?

Reply:
No this is true, the settings we expose are a combination of both protocols, regardless of the semantics there are literally OAuth and OIDC options. This is also based on what we use: next-auth which is self-described as such:

> NextAuth.js is designed to work with any OAuth service, it supports OAuth 1.0, 1.0A, 2.0 and OpenID Connect (OIDC) and has built-in support for most popular sign-in services.